File: movieui/src/models/MovieDetail.ts
import { MovieImage } from './MovieImage';
import { Actor } from './Actor';
import { Category } from './Category';
import { CommentDetail } from './CommentDetail';

export interface MovieDetail {
    id:number,
    name:string,
    mainImage: string;
    imdb:number,
    summary:string,
    releaseDate:string,
    categories:Category[]
    actors:Actor[]
    images:MovieImage[],
    comments:CommentDetail[]
}

File: MovieAPP/Entities/DTOs/MovieDetail.cs
using Core.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entities.DTOs
{
    public class MovieDetail
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Summary { get; set; }
        public string Slug { get; set; }
        public string MainImage { get; set; }
        public decimal Imdb { get; set; }
        public string ReleaseDate { get; set; }
        public List<CommentDetail> Comments { get; set; }
        public List<CategoryDTO> Categories { get; set; }
        public List<ActorDetail> Actors { get; set; }
        public List<MovieImageDetail> Images { get; set; }
    }
}
